Support for multiple object video tracking capabilities with SAM2.1

Buckler89 opened the pull request:
Support for multiple object video tracking capabilities with SAM2

Significant improvements have been made to the video segmentation and tracking features, including support for multi-object tracking.

- Users can now choose between 'box' or 'point' prompts for object identification. Docker support has also been enhanced, with the configuration updated to reserve GPU resources and ensure the correct Docker runtime is used. Additionally, the model configuration and checkpoint paths have been updated for SAM2.1.
- The segment-anything-2 repository can now be installed in any location by setting the appropriate environment variable.
- Lastly, the predict function has undergone major refactoring for improved clarity and efficiency:
- Docker support 
- Support for SAM2.1 models

Known Limitation
- Multi-object tracking is enabled, but due to a bug (#664 ), the UI shows the same label for all objects even though the predictions have different labels.
- Be cautious of the UI, as due to a bug (https://github.com/HumanSignal/label-studio/issues/6593), frames are not displayed properly, causing a misalignment between labels and frames. Under the hood, the labels are still being applied to the correct frames but the UI may not display this correctly.

Thomas-J-Kidd commented:
Im trying to replicate your work! Since you are familiar with this what is the root directory for installing sam2? and also do we need to install it as `sam2` or `segment-anything-model-2`

Buckler89 commented:
Hi @Thomas-J-Kidd 

I've updated the default values of the environment variables to use SAM2.1. If you'd like to use a different model, you can adjust them as needed.

Additionally, I've added documentation to the README about the SEGMENT_ANYTHING_2_REPO_PATH environment variable, which specifies the path to the "segment-anything" repository. You can set it to any location you prefer; personally, I like to install it outside the main repository to avoid having to ignore the folder in the .gitignore file.

Please pull the latest changes to ensure you're up-to-date.

Thanks for your input!

Thomas-J-Kidd commented:
I need to checkout your branch...

anyway now I am running into an error where my tasks variable does not have a key called draft. I am trying to take a look at how the code works but its taking me a while so I thought I would ask!

At the end of the day it comes down to this error:

```python
  File "./mydir/sam-2-test-pr/label-studio-ml-backend/label_studio_ml/examples/segment_anything_2_video/model.py", line 263, in predict
    drafts = tasks[0]['drafts'][0]
             ~~~~~~~~^^^^^^^^^^
KeyError: 'drafts'
```

Thomas-J-Kidd commented:
Thanks for that info! I kept digging last night and found that my context is None. Im still looking into this and it does not seem to track my object regardless of coming back and relabeling. I am not sure what the problem is but I am trying to find out! Hopefully I can pinpoint my error more.

Buckler89 commented:
> Hi @Buckler89 , congrats for the support of SAM2.1 in Label Studio. Just one question, did you manage to study where one could extend Label Studio to have polygons, polylines or masks in video?

Hi @salvaba94, I didn’t handle the polylines because Label Studio doesn’t support that type of annotation yet.
